rust/processor/src/processors/parquet_processors/parquet_fungible_asset_processor.rs

```diff
@@ -0,0 +1,277 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{UploadIntervalConfig, GOOGLE_APPLICATION_CREDENTIALS};
+use crate::{
+    bq_analytics::{
+        create_parquet_handler_loop, generic_parquet_processor::ParquetDataGeneric,
+        ParquetProcessingResult,
+    },
+    db::common::models::{
+        fungible_asset_models::{
+            parquet_coin_supply::CoinSupply,
+            parquet_v2_fungible_asset_balances::FungibleAssetBalance,
+        },
+        object_models::v2_object_utils::{
+            ObjectAggregatedData, ObjectAggregatedDataMapping, ObjectWithMetadata,
+        },
+    },
+    gap_detectors::ProcessingResult,
+    processors::{ProcessorName, ProcessorTrait},
+    utils::{database::ArcDbPool, util::standardize_address},
+};
+use ahash::AHashMap;
+use anyhow::anyhow;
+use aptos_protos::transaction::v1::{write_set_change::Change, Transaction};
+use async_trait::async_trait;
+use chrono::NaiveDateTime;
+use kanal::AsyncSender;
+use serde::{Deserialize, Serialize};
+use std::{fmt::Debug, time::Duration};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct ParquetFungibleAssetProcessorConfig {
+    pub google_application_credentials: Option<String>,
+    pub bucket_name: String,
+    pub bucket_root: String,
+    pub parquet_handler_response_channel_size: usize,
+    pub max_buffer_size: usize,
+    pub parquet_upload_interval: u64,
+}
+
+impl UploadIntervalConfig for ParquetFungibleAssetProcessorConfig {
+    fn parquet_upload_interval_in_secs(&self) -> Duration {
+        Duration::from_secs(self.parquet_upload_interval)
+    }
+}
+
+pub struct ParquetFungibleAssetProcessor {
+    connection_pool: ArcDbPool,
+    coin_supply_sender: AsyncSender<ParquetDataGeneric<CoinSupply>>,
+    fungible_asset_balances_sender: AsyncSender<ParquetDataGeneric<FungibleAssetBalance>>,
+}
+
+impl ParquetFungibleAssetProcessor {
+    pub fn new(
+        connection_pool: ArcDbPool,
+        config: ParquetFungibleAssetProcessorConfig,
+        new_gap_detector_sender: AsyncSender<ProcessingResult>,
+    ) -> Self {
+        if let Some(credentials) = config.google_application_credentials.clone() {
+            std::env::set_var(GOOGLE_APPLICATION_CREDENTIALS, credentials);
```

Comment:
is this something we want to put in each processor or we can move to generic component?